Mathlib/MeasureTheory/Measure/LevyProkhorovMetric.lean

```diff
@@ -387,19 +435,220 @@ lemma continuous_levyProkhorov_to_probabilityMeasure :
+      intro ω ω_in_U
+      simp only [mem_setOf_eq, mem_iUnion, exists_prop] at ω_in_U
+      obtain ⟨k, ⟨B_intersects, _⟩, ω_in_thEk⟩ := ω_in_U
+      rw [mem_thickening_iff] at ω_in_thEk ⊢
+      obtain ⟨w, w_in_Ek, w_near⟩ := ω_in_thEk
+      obtain ⟨z, ⟨z_in_B, z_in_Ek⟩⟩ := nonempty_iff_ne_empty.mpr B_intersects
+      refine ⟨z, z_in_B, lt_of_le_of_lt (dist_triangle ω w z) ?_⟩
+      apply lt_of_le_of_lt (add_le_add w_near.le <|
+              (dist_le_diam_of_mem (Es_bdd k) w_in_Ek z_in_Ek).trans <| Es_diam k)
+      linarith
+
+    -- We use the resulting upper bound `P B ≤ P (Gs JB) + P (small complement)`.
+    apply (measure_mono B_subset).trans
+    apply (measure_union_le _ _).trans
+
+    -- Recall that the small complement is small, `P (small complement) < ε/3`.
+    have aux : P.toMeasure (⋃ j ∈ Iio N, Es j)ᶜ < ENNReal.ofReal (ε/3) := by
+      have rewr : ⋃ i, ⋃ (_ : N ≤ i), Es i = (⋃ i, ⋃ (_ : i < N), Es i)ᶜ :=
+        by simpa only [mem_Iio, compl_Iio, mem_Ici]
+          using (biUnion_compl_eq_of_pairwise_disjoint_of_iUnion_eq_univ
+                  Es_cover Es_disjoint (Iio N)).symm
+      simpa only [mem_Iio, ← rewr] using hN
+
+    -- From the choice of `Q` in a suitable neighborhood, we have `P (Gs JB) < Q (Gs JB) + ε/3`.
+    specialize hQ _ (show JB ⊆ Iio N from fun _ h ↦ h.2)
+
+    -- Now it remains to add the pieces and use the above estimates.
+    apply (add_le_add hQ.le aux.le).trans
+    rw [add_assoc, ← ENNReal.ofReal_add third_ε_pos.le third_ε_pos.le, ← two_mul]
+    apply add_le_add (measure_mono subset_thickB) (ofReal_le_ofReal _)
+    exact δ_gt.le
+
+/-- The topology of the Lévy-Prokhorov metric on probability measures on a separable space
+coincides with the topology of convergence in distribution. -/
+theorem levyProkhorov_eq_convergenceInDistribution :
+    (inferInstance : TopologicalSpace (ProbabilityMeasure Ω))
+      = TopologicalSpace.coinduced (LevyProkhorov.toProbabilityMeasure (Ω := Ω)) inferInstance :=
+  le_antisymm (ProbabilityMeasure.continuous_toLevyProkhorov (Ω := Ω)).coinduced_le
+              levyProkhorov_le_convergenceInDistribution
+
+/-- The identity map is a homeomorphism from `ProbabilityMeasure Ω` with the topology of
+convergence in distribution to `ProbabilityMeasure Ω` with the Lévy-Prokhorov (pseudo)metric. -/
+def homeomorph_probabilityMeasure_levyProkhorov :
+    ProbabilityMeasure Ω ≃ₜ LevyProkhorov (ProbabilityMeasure Ω) where
+  toFun := ProbabilityMeasure.toLevyProkhorov (Ω := Ω)
+  invFun := LevyProkhorov.toProbabilityMeasure (Ω := Ω)
+  left_inv := congrFun rfl
+  right_inv := congrFun rfl
+  continuous_toFun := ProbabilityMeasure.continuous_toLevyProkhorov
+  continuous_invFun := LevyProkhorov.continuous_toProbabilityMeasure
+
+/-- The topology of convergence in distribution on a separable space is pseudo-metrizable. -/
+theorem pseudoMetrizableSpace_probabilityMeasure :
+    PseudoMetrizableSpace (ProbabilityMeasure Ω) :=
```

Comment:
```suggestion
instance : PseudoMetrizableSpace (ProbabilityMeasure Ω) :=
```
Also, in this instance, instead of starting from a pseudometric space, it would be better to start from a pseudometrizable space, I think.

Reply:
I did this, but my proof became very ugly. I kept getting `the synthesized type class is not definitionally equal to ...` until I really had spelled out each and every implicit type class argument in each piece, and in the end I managed to `convert` and rewrite with the equality of the clearly equal topologies...

```
/-- The topology of convergence in distribution on a separable space is pseudo-metrizable. -/
instance (X : Type*)
    [TopologicalSpace X] [PseudoMetrizableSpace X] [SeparableSpace X]
    [MeasurableSpace X] [OpensMeasurableSpace X]:
    PseudoMetrizableSpace (ProbabilityMeasure X) := by
  obtain ⟨psmetr, h⟩ := PseudoMetrizableSpace.exists_pseudo_metric (X := X)
  have opensMble : @OpensMeasurableSpace X psmetr.toUniformSpace.toTopologicalSpace _ := by
    convert ‹OpensMeasurableSpace X›
  have sepble : @SeparableSpace X psmetr.toUniformSpace.toTopologicalSpace := by
    convert ‹SeparableSpace X›
  let homeo := @homeomorph_probabilityMeasure_levyProkhorov X psmetr sepble _ opensMble
  have ind := @Homeomorph.inducing (ProbabilityMeasure X) (LevyProkhorov (ProbabilityMeasure X))
          (@ProbabilityMeasure.instTopologicalSpace X _
            psmetr.toUniformSpace.toTopologicalSpace opensMble) _ homeo
  convert @Inducing.pseudoMetrizableSpace (ProbabilityMeasure X)
          (LevyProkhorov (ProbabilityMeasure X))
          (@ProbabilityMeasure.instTopologicalSpace X _
            psmetr.toUniformSpace.toTopologicalSpace opensMble) _ _ _ ind
  exact h.symm
```

Reply:
If you lookat the docstring of `PseudoMetrizableSpace`, you will see that it gives the advice to use `letI : PseudoMetricSpace X := TopologicalSpace.pseudoMetrizableSpacePseudoMetric X` to endow your space with a metric. With this metric, the topology is defeq to the original one, so all your problems should be solved and the original proof should work just fine.

Reply:
Oh, indeed, thanks a lot!